docs/contributing/performance.rst

```diff
@@ -112,3 +112,63 @@ Here is a rule of thumb to choose among these different options:
 
 Note the design space of the different dimensions is not yet fully explored. We
 may create more specializations in the future when the need arises.
+
+Profiling
+=========
+
+CPU
+---
+
+``perf`` is a commonly available tool for profiling execution on Linux. You can
+collect samples from any execution with ``perf record <command line to
+profile>`` and then use ``perf report`` to view the report.
+
+See `perf examples by Brendan Gregg <https://www.brendangregg.com/perf.html>`_
+for more.
+
+Memory
+------
+
+If you want to profile memory usage, you can enable ``jemalloc`` in the build
+with
+
+.. code-block:: bash
+
+   cmake -DKATANA_USE_JEMALLOC=on <other cmake options>
+
+This will replace the system ``malloc`` implementation with ``jemalloc``, and
+with ``jemalloc``, you can produce heap profiles.
+
+.. warning::
+
+   The standard conda package for jemalloc does not enable profiling. You need
+   to use a conan build instead.
```

Comment:
As we move away from conan and toward using conda more, we will need to fix this. https://katanagraph.atlassian.net/browse/KAT-1387